### PR TITLE
adds shortcuts for aead key and nonce sizes as well as aead ietf vari…

### PR DESCRIPTION
**Pull request by tacitus-aedifex**
…ants

### CODE REVIEW

**Comment by tacitus-aedifex** [comment]
@stef any reason there's no movement on this PR?

**Comment by stef** [comment]
for some reason this is the first time i hear about this PR. will check it within a few days and follow-up with you. thx.

File: test/test_pysodium.py [hunk 1]
```diff
@@ -101,6 +101,16 @@ def test_aead_chacha20poly1305(self):
...
+    def test_aead_chacha20poly1305_ietf(self):
+        key = binascii.unhexlify(b"4290bcb154173531f314af57f3be3b5006da371ece272afa1b5dbdd1100a1007")
+        input_ = binascii.unhexlify(b"86d09974840bded2a5ca")
+        nonce = binascii.unhexlify(b"cd7cf67be39c794a")
+        ad = binascii.unhexlify(b"87e229d4500845a079c0")
+        output = pysodium.crypto_aead_chacha20poly1305_ietf_encrypt(input_, ad, nonce, key)
+        self.assertEqual(binascii.unhexlify(b"e3e446f7ede9a19b62a4677dabf4e3d24b876bb284753896e1d6"), output)
+        output = pysodium.crypto_aead_chacha20poly1305_ietf_decrypt(output, ad, nonce, key)
+        self.assertEqual(output, input_)
+
```
Comment (stef) [inline test/test_pysodium.py]: how did you generate the test vectors? when i run the tests i get even a fail on one of them in line 110, in test_aead_chacha20poly1305_ietf:

    self.assertEqual(binascii.unhexlify(b"e3e446f7ede9a19b62a4677dabf4e3d24b876bb284753896e1d6"), output)
AssertionError: '\xe3\xe4F\xf7\xed\xe9\xa1\x9bb\xa4g}\xab\xf4\xe3\xd2K\x87k\xb2\x84u8\x96\xe1\xd6' != 'l\xa8\x03.\xd2\xd2\x8ae\xdb\x01\xf3\x9aW\xb9\xa9\xa8]n\x9a#\x1dF\xba\xc0\xdet'

File: pysodium/__init__.py [hunk 2]
```diff
@@ -153,6 +157,33 @@ def crypto_aead_chacha20poly1305_decrypt(ciphertext, ad, nonce, key):
     __check(sodium.crypto_aead_chacha20poly1305_decrypt(m, ctypes.byref(mlen), None, ciphertext, clen, ad, adlen, nonce, key))
     return m.raw
 
+# crypto_aead_chacha20poly1305_ietf_encrypt(unsigned char *c, unsigned long long *clen_p, const unsigned char *m, unsigned long long mlen, const unsigned char *ad, unsigned long long adlen, const unsigned char *nsec, const unsigned char *npub, const unsigned char *k)
+def crypto_aead_chacha20poly1305_ietf_encrypt(message, ad, nonce, key):
+
+    if ad == None: raise ValueError("ad cannot be None")
+    mlen = ctypes.c_ulonglong(len(message))
+
+    if ad:
```
Comment (stef) [inline pysodium/__init__.py]: this and the if in 161 is unnecessary there will be an exception anyway on len() i removed this and also in other functions in my branch

**Comment by stef** [comment]
i left out the assertion on the ciphertext, and omitted the checks on the params in this merge, dunno why github doesn't recognize it as a merge. thanks a lot for your kind contribution.
s

**Pull request closed without merging**